File: src/main/java/com/mallang/auth/domain/MemberRepository.java
package com.mallang.auth.domain;

import com.mallang.auth.exception.NotFoundMemberException;
import jakarta.annotation.Nullable;
import org.springframework.data.jpa.repository.JpaRepository;

public interface MemberRepository extends JpaRepository<Member, Long> {

    default Member getByIdIfIdNotNull(@Nullable Long id) {
        if (id == null) {
            return null;
        }
        return getById(id);
    }

    default Member getById(Long id) {
        return findById(id).orElseThrow(NotFoundMemberException::new);
    }
}

File: src/main/kotlin/com/mallang/auth/domain/MemberRepository.kt
package com.mallang.auth.domain

import com.mallang.auth.exception.NotFoundMemberException
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.findByIdOrNull
import java.util.*

interface MemberRepository : JpaRepository<Member, Long> {

    override fun getById(id: Long) = findByIdOrNull(id)
            ?: throw NotFoundMemberException()

    fun findByOauthId(oauthId: OauthId): Optional<Member>
}
